pigeon/android_intents.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:pigeon/pigeon.dart';
+
+// To rebuild this pigeon's output after editing this file,
+// run `tools/check pigeon --fix`.
+@ConfigurePigeon(PigeonOptions(
+  dartOut: 'lib/host/android_intents.dart',
+  kotlinOut: 'android/app/src/main/kotlin/com/zulip/flutter/AndroidIntents.g.kt',
+  kotlinOptions: KotlinOptions(
+    package: 'com.zulip.flutter',
+    // One error class is already generated in AndroidNotifications.g.kt ,
+    // so avoid generating another one, preventing duplicate classes under
+    // the same namespace.
+    includeErrorClass: false)))
+
+// TODO separate out API calls for resolving file name, getting mimetype, getting bytes?
+class IntentSharedFile {
```

Comment:
That sounds potentially useful but definitely a follow-up task, if this version already seems to work correctly :slightly_smiling_face: